roles/openshift_master/tasks/main.yml

```diff
@@ -171,6 +171,14 @@
   when: openshift_master_ha | bool and openshift_master_cluster_method == "native"
 # end workaround for missing systemd unit files
 
+# If the session secrets file was missing from any master, we should
+# clean up the file on masters so that we can get them in sync.
+- name: Delete session secrets file
+  file:
+    path: "{{ openshift.master.session_secrets_file }}"
+    state: absent
+  when: session_secrets_missing | bool
+
```

Comment:
You're right we should be caching these.

On Thursday, December 3, 2015, Jason DeTiberus <notifications@github.com>
wrote:

> In roles/openshift_master/tasks/main.yml
> <https://github.com/openshift/openshift-ansible/pull/1018#discussion_r46619502>
> :
>
> > @@ -171,6 +171,14 @@
> >    when: openshift_master_ha | bool and openshift_master_cluster_method == "native"
> >  # end workaround for missing systemd unit files
> >
> > +# If the session secrets file was missing from any master, we should
> > +# clean up the file on masters so that we can get them in sync.
> > +- name: Delete session secrets file
> > +  file:
> > +    path: "{{ openshift.master.session_secrets_file }}"
> > +    state: absent
> > +  when: session_secrets_missing | bool
> > +
>
> I'm just thinking of the scenario where we start supporting scaling out
> masters (which with native ha *should* be easy), if we re-create the
> secrets on a subset of hosts missing them, then we are going to cause
> client issues on a running system as we scale out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/openshift-ansible/pull/1018/files#r46619502>
> .
>